internal/tool/kubectl.go

```diff
@@ -94,49 +97,76 @@ func NewKubectl(kubeConfig clientcmd.ClientConfig) (*Kubectl, error) {
 	return kubectl, nil
 }
 
-func (k Kubectl) WaitForDeployments(context context.Context, namespace string, selector string) error {
+func (k Kubectl) WaitForWorkloadResources(context context.Context, namespace string, selector string) error {
```

Comment:
I think it would be a good occasion to comment this function

Reply:
comments added to describe the various sections of the function

Reply:
@dcurran90 I think @mgoerens was looking for a top-level comment E.g. 
```suggestion
// WaitForWorkloadResources loops until ...
func (k Kubectl) WaitForWorkloadResources(context context.Context, namespace string, selector string) error {
```

Don't know why that diff looks so awful, but either way, refer to the commenting structure defined here under "Funcs" https://go.dev/doc/comment

Reply:
updated comments.